service/view_building_coordinator.cc

```diff
@@ -299,6 +301,7 @@ future<> view_building_coordinator::send_task(view_building_target target, table
             }
         }
     }
+    _per_host_processing_range.erase(target);
```

Comment:
It makes me uneasy that we are modifying `_per_host_processing_range` without a mutex. I think you should take and hold the group0 client's read apply mutex if you want to touch the variable. Otherwise, it is difficult to reason about what can happen during `get_migrate_tasks_mutations` - you do co_awaits there, so it might change as that function executes.